horizon/facts/opal_forwarder.py

```diff
@@ -61,8 +61,16 @@ def create_data_source_entry(
 
 def create_data_update_entry(entries: list[DataSourceEntry]) -> DataUpdate:
```

Comment:
Better to implement it in `create_data_source_entry()` instead of mutating it after the fact

Reply:
Then we should generate the update id first, then create the data source entries, then create the data update instance. It's not about performance, it's about clarity and logic, making sure the flow makes sense